test/cases/configuration/test_configuration.py

```diff
@@ -208,3 +239,50 @@ def run_precedence_test(self, conf_relative_path):
     def test_datadog_tracing_precedence(self):
         return self.run_precedence_test(
             "./conf/datadog_tracing_precedence.conf")
+
+    def test_datadog_unified_service_tagging_precedence(self):
+        conf_path = (Path(__file__).parent / "conf" /
+                     "datadog_unified_service_tagging.conf")
+        conf_text = conf_path.read_text()
+        status, log_lines = self.orch.nginx_replace_config(
+            conf_text, conf_path.name)
+        self.assertEqual(status, 0, log_lines)
+
+        status, _, body1 = self.orch.send_nginx_http_request("/http", port=80)
+        self.assertEqual(status, 200)
+
+        status, _, body2 = self.orch.send_nginx_http_request("/http", port=81)
+        self.assertEqual(status, 200)
+
+        status, _, body3 = self.orch.send_nginx_http_request("/http", port=82)
+        self.assertEqual(status, 200)
+
+        config_srv1 = json.loads(body1)
+        pattern = {
+            "service": "http_block",
+            "environment": "http",
+            "version": "1.5.0-http",
+        }
+
+        mismatches = find_mismatches(pattern, config_srv1)
+        self.assertEqual(mismatches, [])
+
+        config_srv2 = json.loads(body2)
+        pattern = {
+            "service": "server1_block",
+            "environment": "server1",
+            "version": "1.5.0-server1",
+        }
+
+        mismatches = find_mismatches(pattern, config_srv2)
+        self.assertEqual(mismatches, [])
+
+        config_srv3 = json.loads(body3)
+        pattern = {
```

Comment:
Can we update this test case so that it only partially overrides the settings? Perhaps only update `datadog_service_name` and validate the output of that

Reply:
👍

Reply:
This test assess the precedence of unified service tagging directives, rather than just the `datadog_service_name`. Partially overriding only `datadog_service_name` would defeat the purpose of the test. Am I misunderstanding something?

Reply:
My +1 was on the wrong comment (wanted to write here). I think what he meant was that it would be interesting that partial modification does not break the rest of the serv_conf.